src/main/java/com/google/starfish/servlets/HandleNotesServlet.java

```diff
@@ -1,58 +1,128 @@
       return;
     }
 
     HttpSession activeSession = request.getSession(false);
     String userId = (String) activeSession.getAttribute("user_id");
-
     String blobKey = getUploadedFileBlobKey(request, "file");
-    if(blobKey == null) {
+    String sourceUrl = request.getParameter("sourceUrl");
+
+    // There must be some source, either a blob or url, for this note to be posted
+    if (blobKey == null && sourceUrl == null) {
       response.setStatus(HttpServletResponse.SC_NOT_ACCEPTABLE);
       return;
     }
-    String title = request.getParameter("title").toLowerCase();
+
+    String title = request.getParameter("title");
     String school = request.getParameter("school").toLowerCase();
     String course = request.getParameter("course").toLowerCase();
-    String[] miscLabels = request.getParameter("miscLabels").split(",");
-
-    // TODO: Put labels into database
+    String[] miscLabels = request.getParameterValues("miscLabels");
 
-    // TODO: Put note into database
+    DataSource pool = (DataSource) request.getServletContext().getAttribute("my-pool");
 
-    response.setStatus(HttpServletResponse.SC_OK);
+    // Insert all labels before inserting note due to foreign key constraint
+    labelService.insertSchoolLabel(pool, school);
+    labelService.insertCourseLabel(pool, course);
+    for (String miscLabel : miscLabels) {
+      labelService.insertMiscLabel(pool, miscLabel.toLowerCase());
+    }
+    
+    try (Connection conn = pool.getConnection()) {
+      String stmt =
+          "INSERT INTO notes ( "
+              + "author_id,"
+              + "school,"
+              + "course,"
+              + "title,"
+              + "source_url,"
+              + "pdf_source,"
+              + "date_created,"
+              + "num_downloads ) "
+        + "VALUES ( "
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "? ); ";
+
+      try (PreparedStatement noteStmt = conn.prepareStatement(stmt, new String[] {"id"})) {
+        noteStmt.setString(1, userId);
+        noteStmt.setString(2, school);
+        noteStmt.setString(3, course);
+        noteStmt.setString(4, title);
+        if (sourceUrl == null) {
+          // If there was no sourceUrl, this is an uploaded pdf
+          // so set the source url to serve-notes path
+          noteStmt.setString(5, "/serve-notes?key=" + blobKey);
+        } else {
+          // If there is a sourceUrl on the request param, this is
+          // a google doc set set the source url based on the passed param
+          noteStmt.setString(5, sourceUrl);
+        }
+        // The blob key is nullable on the notes table so if this is a google doc
+        // this will be null
+        noteStmt.setString(6, blobKey);
+        noteStmt.setDate(7, new Date(Calendar.getInstance().getTimeInMillis()));
+        noteStmt.setLong(8, 0); // num_downloads
```

Comment:
if numdownloads is always 0, we don't need to dynamically set it here, we can just set it in the prepared statement.

Reply:
Good point, I'll make that change